activestorage/lib/active_storage/attached/model.rb

```diff
@@ -147,7 +147,7 @@ def #{name}=(attachables)
             else
               if Array(attachables).any?
                 attachment_changes["#{name}"] =
-                  ActiveStorage::Attached::Changes::CreateMany.new("#{name}", self, #{name}.blobs + attachables)
+                  ActiveStorage::Attached::Changes::CreateMany.new("#{name}", self, #{name}.persisted_blobs + attachables)
```

Comment:
just curious, why filtering for persisted blobs here instead of in `Attached::Changes::CreateMany#blobs` ?

Reply:
> user = User.new(name: "Tina", highlights: [fixture_file_upload("image.gif")])
The above line would process and associate a new blob(`image.gif`) to the highlights attachment and `attachables` would also hold `image.gif` as transaction is not complete.

> user.highlights.attach*fixture_file_upload("racecar.jpg"))
This would append `racecar.jpg` to the attachables and since we are doing `highlights.blobs`, the new blob(`image.gif`) would be appended again. Filtering the unpersisted blob here removes the chance of duplicate addition.

This is the reasons files get duplicated when used multiple Active Storage setter method in one transaction.

Reply:
I meant that filtering the persisted blobs in the `blobs` method in the `Attached::Changes::CreateMany` class seems to fix this as well. My question was if there was a reason why doing the fix in the setter instead of in the `Attached::Changes::CreateMany` class. But it was just out of curiosity, I don't have any reason why to do one or the other thing

Reply:
I also wonder if it would be better to move this logic into `Attached::Changes::CreateMany`, for example there is some possibly related logic here:

https://github.com/rails/rails/blob/c9a0f1ab9616ca8e94f03327259ab61d22f04b51/activestorage/lib/active_storage/attached/changes/create_many.rb#L47-L49

Take this for what you will though, it's probably better to get feedback from @pixeltrix in this case.